### PR TITLE
Catch fault when accessing Document node with XPath

### LINKED ISSUE #103: Selecting root with Xpath directive does not work
### Bug Report or Feature Request (mark with an x)
- [x] bug report -> please search issues before submitting
- [ ] feature request

### Expected Behavior
I can add a child by selecting the root using this:
```new Directives().Add("root").Add("child").Xpath("/").Attr("some","attribute")```

### Actual Behavior
Exception is thrown:

```
Yaapii.Xml.Xambly.ImpossibleModificationException : Exception at dir 4: ATTR 'name', 'repo-2.xml' (Unable to cast object of type 'System.Xml.XmlDocument' to type 'System.Xml.XmlElement'.)
```

### PR DESCRIPTION
**Pull request by aveXcaesar**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)
#103 

### What is the new behavior?
Useful exception message with hint to solve the fault

### Does this PR introduce a breaking change? (check one with "x")
- [ ] Yes
- [x] No

###  Other information
closes #103

### CODE REVIEW

File: tests/Yaapii.Xml.Xambly.Tests/Directive/DirectivesTest.cs [hunk 1]
```diff
@@ -288,6 +288,31 @@ public void ParsesGrammar()
...
+                var xml = new XmlDocument();
+                var xambler =
+                    new Xambler(
+                        new Directives()
+                            .Add("root")
+                            .Add("child")
+                            .Xpath("/")
+                            .Attr("some", "attribute")
+                    ).Apply(xml);
+                Assert.True(false, "Exception expected!");
```
Comment (koeeenig) [inline tests/Yaapii.Xml.Xambly.Tests/Directive/DirectivesTest.cs]: Please use `Assert.Throws` from xunit framework.

Comment (aveXcaesar) [reply tests/Yaapii.Xml.Xambly.Tests/Directive/DirectivesTest.cs]: `Assert.Throws` doesn't give the possibility to checks exception's message. If a implement that I can check the existence of an exception but not it's content.
I would prefer to let it as it is.

Comment (koeeenig) [reply tests/Yaapii.Xml.Xambly.Tests/Directive/DirectivesTest.cs]: Maybe we should throw a more typed exception? Relying on the message content of an exception fells kind da wrong to me. What do you think @Meerownymous ?

Comment (Meerownymous) [reply tests/Yaapii.Xml.Xambly.Tests/Directive/DirectivesTest.cs]: Agree with @koeeenig, it should be tested if the correct Exception type is throw . If you want to check the message, this is not the right place - that should be done in a test which tests the Exception class. message is for context, type is for application flow.

Comment (aveXcaesar) [reply tests/Yaapii.Xml.Xambly.Tests/Directive/DirectivesTest.cs]: Okay, I will change it.
But an open question for me: How test the exception message?
@Meerownymous: You say: "that should be done in a test which tests the Exception class"
The message is generated in an class which uses the exception (here `Directives()`). Then I haven't the chance to test the message content in an exception test.
Do we need an new different `Assert.Throws()` method which tests the content of an exception message?

Comment (Meerownymous) [reply tests/Yaapii.Xml.Xambly.Tests/Directive/DirectivesTest.cs]: I simply think, testing the message for each class which rises the exception is kind of over-engineered. Lets test the type, and maybe on the other hand if the exception class is built correctly, but thats it.

Comment (aveXcaesar) [reply tests/Yaapii.Xml.Xambly.Tests/Directive/DirectivesTest.cs]: okay

**Comment by aveXcaesar** [comment]
Changes done

**Pull request merged**